docs/en/operation/StressBench.md

```diff
@@ -0,0 +1,76 @@
+---
+layout: global
+title: StressBench Framework
+nickname: StressBench
+group: Operations
+priority: 7
+---
+
+* Table of Contents
+  {:toc}
+
+The Alluxio system runs in different environments with different hardware for various purposes. The
+StressBench test suite is a performance benchmark suite to help you understand the performance of
+Alluxio in different scenarios. It will help you quickly test and analyze the Alluxio usage in your
+environment by simply running the command line.
+
+The StressBench is consisted of different tests to help you test the metadata and IO performance of
+the Alluxio system. The following benchmark tools are currently supported:
+* `StressMasterBench` - A benchmark tool measuring the master performance of Alluxio.
+* `StressClientBench` - A benchmark tool measuring the IO performance of Alluxio through the client.
+* `StressWorkerBench` - A benchmark tool measuring the IO performance of Alluxio with single node.
+* `UfsIOBench` - A benchmark tool measuring the IO performance to UFS.
+* `FuseIOBench` - A benchmark tool measuring the IO performance of Alluxio through Fuse interface.
+
+## High-Level Design
+StressBench is a job that runs with the job service. Therefore, the design follows what the job
+service supports. There are a few components for StressBench as follows:
+
+### Client CLI
+The user-facing CLI program allows you to invoke the StressBench tasks through the command line.
+
+### Job Definition
+This is the definition of the job master, which is invoked by the client CLI program, and will
+further coordinate distributed tasks for the job.
+
+### Distributed Tasks
+These tasks are run on the job worker, and eventually report back to the job master.
+
+## Consistent Results
+When testing results are inconsistent and wildy variable, it makes trusting the results more
+difficult. Therefore, there are some features that help StressBench to be consistent with results.
+
+### Barriers
+Barriers are one way that StressBench tries to achieve consistent results. This allows distributed
```

Comment:
How are the start time determined? I feel like using a start time heuristically does not give something equivalent to what we mean typically by "Barrier" with signals and conditions? (I'm not sure how to improve the description here).

Reply:
https://en.wikipedia.org/wiki/Barrier_(computer_science) barrier is a fair statement here.
You can check https://github.com/Alluxio/alluxio/blob/v2.4.1/stress/shell/src/main/java/alluxio/stress/cli/StressMasterBench.java#L224 to see how we define the timestamp of when the tasks should start in `StressMasterBench`
, and https://github.com/Alluxio/alluxio/blob/v2.4.1/job/server/src/main/java/alluxio/job/plan/stress/StressBenchDefinition.java#L161 for `start time`.